lib/rspec-puppet/support.rb

```diff
@@ -279,6 +279,34 @@ def trusted_facts_hash(node_name)
       extensions
     end
 
+    def server_facts_hash
+      server_facts = {}
+
+      # Add our server version to the fact list
+      server_facts["serverversion"] = Puppet.version.to_s
+
+      # And then add the server name and IP
+      {"servername" => "fqdn",
+        "serverip" => "ipaddress"
+      }.each do |var, fact|
+        if value = Facter.value(fact)
+          server_facts[var] = value
+        else
+          warn "Could not retrieve fact #{fact}"
+        end
+      end
+
+      if server_facts["servername"].nil?
+        host = Facter.value(:hostname)
+        if domain = Facter.value(:domain)
+          server_facts["servername"] = [host, domain].join(".")
+        else
+          server_facts["servername"] = host
+        end
```

Comment:
It seems like there should be a check here to verify that a complete `$server_facts` hash was created (e.g., with all four keys and self-consistent values).

A partially-complete hash could cause tests for correct Puppet code to fail (which appears to be happening in at least [some of the Travis tests](https://travis-ci.org/rodjek/rspec-puppet/jobs/332177211#L5473)).